packages/block-editor/src/components/inserter/block-patterns-tab.js

```diff
@@ -208,21 +165,15 @@ export function BlockPatternsCategoryPanel( {
 
 				return availablePatternCategories.length === 0;
 			} ),
-		[ allPatterns, category ]
+		[ allPatterns, availableCategories, category.name ]
 	);
-	const patterns =
-		category.name === 'reusable'
-			? filteredUnsyncedPatterns
-			: currentCategoryPatterns;
-	const currentShownPatterns = useAsyncList( patterns );
+
+	const categoryPatternsList = useAsyncList( currentCategoryPatterns );
 
 	// Hide block pattern preview on unmount.
-	useEffect( () => () => onHover( null ), [] );
+	useEffect( () => () => onHover( null ), [ onHover ] );
```

Comment:
This might be a bit dangerous since we didn't require `onHover` to be a stable function instance. Which means someone might accidentally pass in an unmemoized function and trigger this effect every render.

We might want to:
1. Use a ref to create a (semi-)stable reference to the function
2. Ignore the eslint warning and add comment above explaining why.